zio-http/shared/src/main/scala/zio/http/Body.scala

```diff
@@ -578,13 +578,18 @@ object Body {
             (fs, size) = r
```

Comment:
Maybe we should change this to `suspendSucceed` instead and use a try-catch block to avoid the additional fold?

Reply:
@kyri-petrou is this what you had in mind: https://github.com/zio/zio-http/pull/3234/files? 🤔